rtrlib/rtr_mgr.c

```diff
@@ -431,17 +507,148 @@ inline int rtr_mgr_get_spki(struct rtr_mgr_config *config,
 			    struct spki_record **result,
 			    unsigned int *result_count)
 {
-	return spki_table_get_all(config->groups[0].sockets[0]->spki_table,
+	return spki_table_get_all(config->spki_table,
 				  asn, ski, result, result_count);
 }
 
 void rtr_mgr_stop(struct rtr_mgr_config *config)
 {
+	tommy_node *node = tommy_list_head(&config->groups);
+	struct rtr_mgr_group_node *group_node;
+
 	MGR_DBG1("rtr_mgr_stop()");
-	for (unsigned int i = 0; i < config->len; i++) {
-		for (unsigned int j = 0; j < config->groups[i].sockets_len; j++)
-			rtr_stop(config->groups[i].sockets[j]);
+	while (node) {
+		group_node = node->data;
+		for (unsigned int j = 0; j < group_node->group->sockets_len;
+		     j++) {
+			rtr_stop(group_node->group->sockets[j]);
+		}
+	node = node->next;
+	}
+}
+
+int rtr_mgr_add_group(struct rtr_mgr_config *config,
+		      const struct rtr_mgr_group *group)
+{
+	unsigned int refresh_iv = 3600;
+	unsigned int retry_iv = 600;
+	unsigned int expire_iv = 7200;
+	int err_code = RTR_ERROR;
```

Comment:
In #98 there are default values for these intervals. Could they serve the purpose for this problem?

```
static const uint16_t RTR_EXPIRATION_DEFAULT = 7200;
// ...
static const uint16_t RTR_REFRESH_DEFAULT = 3600;
// ...
static const uint16_t RTR_RETRY_DEFAULT = 600;
```